crates/electrum/src/bdk_electrum_client.rs

```diff
@@ -276,7 +276,13 @@ impl<E: ElectrumApi> BdkElectrumClient<E> {
 
                 for tx_res in spk_history {
                     tx_update.txs.push(self.fetch_tx(tx_res.tx_hash)?);
-                    self.validate_merkle_for_anchor(tx_update, tx_res.tx_hash, tx_res.height)?;
+                    if tx_res.height > 0 {
+                        self.validate_merkle_for_anchor(
+                            tx_update,
+                            tx_res.tx_hash,
+                            tx_res.height.try_into().expect("must not overflow"),
+                        )?;
```

Comment:
:pick: We can just match on the try into result and branch based on that to avoid the unwrap/expect.